src/main/java/com/epam/ta/reportportal/core/item/impl/DeleteTestItemHandlerImpl.java

```diff
@@ -52,26 +53,41 @@
 @Service
 class DeleteTestItemHandlerImpl implements DeleteTestItemHandler {
 
-	private TestItemRepository testItemRepository;
+	private final TestItemRepository testItemRepository;
 
-	// TODO ANALYZER
-	//	@Autowired
-	//	private ILogIndexer logIndexer;
+	private final LaunchRepository launchRepository;
 
 	@Autowired
-	public void setTestItemRepository(TestItemRepository testItemRepository) {
+	public DeleteTestItemHandlerImpl(TestItemRepository testItemRepository, LaunchRepository launchRepository) {
 		this.testItemRepository = testItemRepository;
+		this.launchRepository = launchRepository;
 	}
 
+	// TODO ANALYZER
+	//	@Autowired
+	//	private ILogIndexer logIndexer;
+
 	@Override
 	public OperationCompletionRS deleteTestItem(Long itemId, ReportPortalUser.ProjectDetails projectDetails,
 			ReportPortalUser reportPortalUser) {
 		TestItem item = testItemRepository.findById(itemId)
 				.orElseThrow(() -> new ReportPortalException(ErrorType.TEST_ITEM_NOT_FOUND, itemId));
-		validate(item, reportPortalUser, projectDetails);
+
+		Optional<TestItem> retryParent = ofNullable(item.getRetryOf()).map(retryParentId -> testItemRepository.findById(retryParentId)
+				.orElseThrow(() -> new ReportPortalException(ErrorType.TEST_ITEM_NOT_FOUND, retryParentId)));
+
+		Launch launch = retryParent.map(TestItem::getLaunch).orElseGet(item::getLaunch);
+
+		validate(item, launch, reportPortalUser, projectDetails);
 		Optional<TestItem> parent = ofNullable(item.getParent());
-		testItemRepository.delete(item);
 
+		retryParent.ifPresent(rp -> rp.getRetries().remove(item));
```

Comment:
we can't delete retry items
they should be deleted only when parent is

Reply:
fixed